### PR TITLE
Remove Dsl constructor from plugin context

### PR DESCRIPTION
**Pull request by Aschen**
#  What does this PR do ?

Remove the `Dsl` deprecated property from the plugin context.

### Other changes

 - allow `server:publicApi` instead of `server:info` in standard roles
 - Reset cached profiles and roles when calling `admin:resetSecurity`
 - Add page for Docker migration guide
 - fix functional tests again :upside_down_face:

### CODE REVIEW

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [adminController.js#L57](https://github.com/kuzzleio/kuzzle/blob/6ccdb9e5bf7c0dd4b7735593d8e0684e77082115/lib/api/controllers/adminController.js#L57): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [adminController.js#L57](https://github.com/kuzzleio/kuzzle/blob/0542d5dd13eaca0c30642fc1b27b6a974c3c487e/lib/api/controllers/adminController.js#L57): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [adminController.js#L57](https://github.com/kuzzleio/kuzzle/blob/1c1e69990f04764ee6e4b57250eee867d396a7c3/lib/api/controllers/adminController.js#L57): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [adminController.js#L57](https://github.com/kuzzleio/kuzzle/blob/6cf1af24f7e84bf13047a22cf74060b8ff74cda2/lib/api/controllers/adminController.js#L57): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [pluginContext.js#L95](https://github.com/kuzzleio/kuzzle/blob/6cf1af24f7e84bf13047a22cf74060b8ff74cda2/lib/api/core/plugins/pluginContext.js#L95): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [adminController.js#L57](https://github.com/kuzzleio/kuzzle/blob/b91284bfd9817e13b5592c7d40a0a8522b8ace87/lib/api/controllers/adminController.js#L57): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [pluginContext.js#L95](https://github.com/kuzzleio/kuzzle/blob/b91284bfd9817e13b5592c7d40a0a8522b8ace87/lib/api/core/plugins/pluginContext.js#L95): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 6 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [adminController.js#L57](https://github.com/kuzzleio/kuzzle/blob/02cecd735d75f9103fd02c04f4a9013d86144c55/lib/api/controllers/adminController.js#L57): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [repository.js#L69](https://github.com/kuzzleio/kuzzle/blob/02cecd735d75f9103fd02c04f4a9013d86144c55/lib/api/core/models/repositories/repository.js#L69): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [pluginContext.js#L95](https://github.com/kuzzleio/kuzzle/blob/02cecd735d75f9103fd02c04f4a9013d86144c55/lib/api/core/plugins/pluginContext.js#L95): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [elasticsearch.js#L1523](https://github.com/kuzzleio/kuzzle/blob/02cecd735d75f9103fd02c04f4a9013d86144c55/lib/services/elasticsearch.js#L1523): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [elasticsearch.js#L1564](https://github.com/kuzzleio/kuzzle/blob/02cecd735d75f9103fd02c04f4a9013d86144c55/lib/services/elasticsearch.js#L1564): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [elasticsearch.js#L1642](https://github.com/kuzzleio/kuzzle/blob/02cecd735d75f9103fd02c04f4a9013d86144c55/lib/services/elasticsearch.js#L1642): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 6 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [adminController.js#L57](https://github.com/kuzzleio/kuzzle/blob/1ccb31576c0326a3d26d343dd17cecb7e428bccd/lib/api/controllers/adminController.js#L57): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [repository.js#L69](https://github.com/kuzzleio/kuzzle/blob/1ccb31576c0326a3d26d343dd17cecb7e428bccd/lib/api/core/models/repositories/repository.js#L69): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [pluginContext.js#L95](https://github.com/kuzzleio/kuzzle/blob/1ccb31576c0326a3d26d343dd17cecb7e428bccd/lib/api/core/plugins/pluginContext.js#L95): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [elasticsearch.js#L1523](https://github.com/kuzzleio/kuzzle/blob/1ccb31576c0326a3d26d343dd17cecb7e428bccd/lib/services/elasticsearch.js#L1523): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [elasticsearch.js#L1564](https://github.com/kuzzleio/kuzzle/blob/1ccb31576c0326a3d26d343dd17cecb7e428bccd/lib/services/elasticsearch.js#L1564): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [elasticsearch.js#L1642](https://github.com/kuzzleio/kuzzle/blob/1ccb31576c0326a3d26d343dd17cecb7e428bccd/lib/services/elasticsearch.js#L1642): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: doc/2/guides/upgrade-kuzzle/changes/index.md [hunk 1]
```diff
@@ -160,6 +182,8 @@ It accepts the same arguments as the `kuzzle start` command from the CLI.
 
   - `server.protocols.socketio.enable` is now `false`, deactivating the Socket.io protocol by default
   - `services.storage.commonMapping.dynamic` is now `false` by default, meaning that Elasticsearch will not infer mapping of new introduced fields
+  - `security.standard.roles.default.controllers.server.actions` is now `{ publicApi: true }` instead of `{ info: true }`
+  - `security.standard.roles.anonymous.controllers.server.actions` is now `{ publicApi: true }` instead of `{ info: true }`
```
Comment (scottinet) [inline doc/2/guides/upgrade-kuzzle/changes/index.md]: This should be added to the upgrade script: https://github.com/kuzzleio/kuzzle/blob/2-dev/bin/.upgrades/versions/v1/checkConfiguration.js#L28

File: doc/2/guides/upgrade-kuzzle/changes/index.md [hunk 2]
```diff
@@ -188,3 +212,22 @@ To:
 ### Plugins
 
   - Plugins manifest files are now required
+  - `Dsl` constructor from the plugin context is now removed, use `Koncorde` instead (deprecated in 1.4.0)
+
+## Docker images
+
+## Kuzzle
+
+Kuzzle images are now built for the two major version of Kuzzle.
```
Comment (scottinet) [inline doc/2/guides/upgrade-kuzzle/changes/index.md]:
```suggestion
Kuzzle images are now built for the two major versions of Kuzzle.  
```

File: doc/2/guides/upgrade-kuzzle/changes/index.md [hunk 3]
```diff
@@ -188,3 +212,22 @@ To:
...
   - Plugins manifest files are now required
+  - `Dsl` constructor from the plugin context is now removed, use `Koncorde` instead (deprecated in 1.4.0)
+
+## Docker images
+
+## Kuzzle
+
+Kuzzle images are now built for the two major version of Kuzzle.  
+
+This includes `kuzzleio/kuzzle` production image but also the `kuzzleio/plugin-dev` image for plugin developers.
```
Comment (scottinet) [inline doc/2/guides/upgrade-kuzzle/changes/index.md]:
```suggestion
This includes the `kuzzleio/kuzzle` production image but also the `kuzzleio/plugin-dev` image for plugin developers.  
```

File: doc/2/guides/upgrade-kuzzle/changes/index.md (hunk 1 above)
Comment (Aschen) [reply doc/2/guides/upgrade-kuzzle/changes/index.md]: Why ? I didn't rename the entry, I change the value of default rights

**Comment by kuzzle** [comment]
SonarQube analysis reported 6 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [adminController.js#L57](https://github.com/kuzzleio/kuzzle/blob/ac3ad9a1f661b4d4af638fbbeb1941b2382cb1fb/lib/api/controllers/adminController.js#L57): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [repository.js#L69](https://github.com/kuzzleio/kuzzle/blob/ac3ad9a1f661b4d4af638fbbeb1941b2382cb1fb/lib/api/core/models/repositories/repository.js#L69): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [pluginContext.js#L95](https://github.com/kuzzleio/kuzzle/blob/ac3ad9a1f661b4d4af638fbbeb1941b2382cb1fb/lib/api/core/plugins/pluginContext.js#L95): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [elasticsearch.js#L1523](https://github.com/kuzzleio/kuzzle/blob/ac3ad9a1f661b4d4af638fbbeb1941b2382cb1fb/lib/services/elasticsearch.js#L1523): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [elasticsearch.js#L1564](https://github.com/kuzzleio/kuzzle/blob/ac3ad9a1f661b4d4af638fbbeb1941b2382cb1fb/lib/services/elasticsearch.js#L1564): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [elasticsearch.js#L1642](https://github.com/kuzzleio/kuzzle/blob/ac3ad9a1f661b4d4af638fbbeb1941b2382cb1fb/lib/services/elasticsearch.js#L1642): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: doc/2/guides/upgrade-kuzzle/changes/index.md (hunk 1 above)
Comment (scottinet) [reply doc/2/guides/upgrade-kuzzle/changes/index.md]: isn't `security.standard.roles.anonymous.controllers.server.actions.info` obsolete now?

Comment (scottinet) [reply doc/2/guides/upgrade-kuzzle/changes/index.md]: Ah, my bad.

**Pull request merged**